src/functions.php

```diff
@@ -31,6 +46,14 @@ function reject($promiseOrValue = null)
         });
     }
 
+    if ($promiseOrValue instanceof AsyncInteropPromise) {
+        return new Promise(function ($resolve, $reject) use ($promiseOrValue) {
+            $promiseOrValue->when(function ($reason = null, $value = null) use ($resolve, $reject) {
+                $reject($reason ? $reason : $value);
```

Comment:
Yes, we allow rejection without a reason:
https://github.com/reactphp/promise/blob/2.x/src/functions.php#L26
https://github.com/reactphp/promise/blob/2.x/src/RejectedPromise.php#L9